packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1526,10 +1539,13 @@ export class EthModule {
     return bufferToRpcData(tx.hash());
   }
 
+  /**
+   * Returns the trace of the sent tx
+   */
   private async _handleMineBlockResults(
     results: MineBlockResult[],
     sentTx: TypedTransaction
-  ) {
+  ): Promise<GatherTracesResult> {
```

Comment:
I added this to handle the error-throwing logic in the callee. That way, adding the extra fields to the error is done in a single place. Otherwise the error would've been modified both here and in a `catch` in the callee.